jsinterop-ts-defs-processor/src/main/java/com/vertispan/tsdefs/model/TsType.java

```diff
@@ -59,10 +59,14 @@ public TsType addBounds(TsType tsType) {
   }
 
   public String emit(String parentNamespace) {
+    return emitType(parentNamespace) + emitNullable();
+  }
+
+  protected String emitType(String parentNamespace) {
     if (namespace.isEmpty() || namespace.equals(parentNamespace)) {
-      return resolveName(name) + emitBounds(parentNamespace) + emitNullable();
+      return resolveName(name) + emitBounds(parentNamespace);
     } else {
-      return namespace + "." + resolveName(name) + emitBounds(parentNamespace) + emitNullable();
+      return namespace + "." + resolveName(name) + emitBounds(parentNamespace);
```

Comment:
I noted this in the ticket, but I think that arrays need to be adjusted as well.